x-pack/plugins/observability/public/pages/slos/components/slo_list_search_filter_sort_bar.tsx

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFieldSearch,
+  EuiFilterButton,
+  EuiFilterGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiSelectable,
+  EuiSelectableOption,
+} from '@elastic/eui';
+import { EuiSelectableOptionCheckedType } from '@elastic/eui/src/components/selectable/selectable_option';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+
+export interface SloListSearchFilterSortBarProps {
+  loading: boolean;
+  onChangeQuery: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onChangeSort: (sortMethod: SortType) => void;
+  onChangeStatusFilter: (statusFilters: SortItem[]) => void;
+}
+
+export type SortType = 'difference' | 'budgetRemaining';
+export type StatusType = 'violated' | 'forecastedViolation' | 'healthy';
```

Comment:
For SLO defined as calendar aligned and timeslices, we might need to handle a fourth status: "degrading" (or whatever better name we come up with), when the observed value is below the objective but the error budget has some room left.

For rolling SLO, this can never happen, if we are below the objective, it means we have consumed all the error budget.

For occurrences and calendar aligned SLO, we are indeed in forecasted violation.